File: C++/Rebill/Create_Recurring_Payment_ACH.cpp
//
// BluePay C++ Sample code.
//
// This code sample creates a recurring payment charging $15.00 per month for one year. 

#include "Create_Recurring_Payment_ACH.h"
#include "../bluepay-cpp/BluePay.h"
#include <iostream>
using namespace std;

void createRecurringPaymentACH(){
    string accountId = "Merchant's Account ID Here";
    string secretKey = "Merchant's Secret Key Here";
    string mode = "TEST";

    BluePay rebill(
    	accountId, 
	    secretKey, 
	    mode 
    );

    rebill.setCustomerInformation(
	    "Bob", // First Name
	    "Tester", // Last Name
	    "123 Test St.", // Address1
	    "Apt #500", // Address2
	    "Testville", // City
	    "IL", // State
	    "54321", // Zip
	    "USA", // Country
	    "1231231234", // Phone Number
	    "test@bluepay.com" // Email Address
	  );

    rebill.setACHInformation(
    	"123123123", // Routing Number
    	"0523421", // Account Number
    	"C", // Account Type: Checking
    	"WEB" // ACH Document Type: WEB
    );

	rebill.setRebillingInformation(
		"15.00", // Rebill Amount
		"2015-01-01", // Rebill Start Date
		"1 MONTH", // Rebill Frequency
		"12" // Rebill # of Cycles
	);

	// Auth Amount: $0.00
	rebill.auth("0.00");
    
    // Makes the API Request with Blue
	rebill.process();

	// Reads the responses from BluePay if transaction was approved
	if (rebill.isSuccessfulTransaction()){
	    cout << string("Transaction ID: ") + rebill.getTransId() + "\n";
		cout << string("Rebill ID: ") + rebill.getRebillId() + "\n";
	    cout << string("Transaction Status: ") + rebill.getResult() + "\n";
	    cout << string("Transaction Message: ") + rebill.getMessage() + "\n";
	    cout << string("Masked Payment Account: ") + rebill.getMaskedPaymentAccount() + "\n";
	    cout << string("Bank Name: ") + rebill.getBank() + "\n";
	}
	else {
	    cout << string("Error: ") + rebill.getMessage();
	}	
}

File: C#/Rebill/Create_Recurring_Payment_ACH.cs
/*
* BluePay C#.NET Sample code.
*
* This code sample creates a recurring payment charging $15.00 per month for one year.
*/

using System;
using BluePayLibrary;

namespace Rebill
{
	class CreateRecurringPaymentACH
	{
		public static void Main ()
		{
			string accountID = "Merchant's Account ID Here";
			string secretKey = "Merchant's Secret Key Here";
			string mode = "TEST";

			BluePay rebill = new BluePay
			(
			    accountID,
			    secretKey,
			    mode
			);

			rebill.SetCustomerInformation
			(
			    firstName: "Bob",
			    lastName: "Tester",
			    address1: "1234 Test St.",
			    address2: "Apt #500",
			    city: "Testville",
			    state: "IL",
			    zip: "54321",
			    country: "USA",
			    phone: "123-123-12345",
			    email: "test@bluepay.com"
			);

			rebill.SetACHInformation
			(
			    routingNum: "123123123",
			    accountNum: "123456789",
			    accountType: "C",
			    docType: "WEB"
			);

			// Sets recurring payment
			rebill.SetRebillingInformation
			(
			    rebFirstDate: "2015-01-01", // Rebill Start Date: Jan. 1, 2015
			    rebExpr: "1 MONTH", // Rebill Frequency: 1 MONTH
			    rebCycles: "12", // Rebill # of Cycles: 12
			    rebAmount: "15.00" // Rebill Amount: $15.00
			);

			// Sets a Card Authorization at $0.00
			rebill.Auth(amount: "0.00");

			// Makes the API Request
			rebill.Process();

			// If transaction was successful reads the responses from BluePay
			if (rebill.IsSuccessfulTransaction())
			{
			    Console.WriteLine("Transaction ID: " + rebill.GetTransID());
			    Console.WriteLine("Rebill ID: " + rebill.GetRebillID());
			    Console.WriteLine("Transaction Status: " + rebill.GetStatus());
			    Console.WriteLine("Transaction Message: " + rebill.GetMessage());
			    Console.WriteLine("Masked Payment Account: " + rebill.GetMaskedPaymentAccount());
			    Console.WriteLine("Customer Bank Name: " + rebill.GetBank());
			}
			else
			{
			    Console.WriteLine("Error: " + rebill.GetMessage());
			}
		}
	}
}

